pennylane_lightning/src/util/Util.hpp

```diff
@@ -334,7 +336,7 @@ auto transpose_state_tensor(const std::vector<T> &tensor,
     -> std::vector<T> {
     std::vector<T> transposed_tensor(tensor.size());
     for (size_t ind = 0; ind < tensor.size(); ind++) {
-        transposed_tensor[transposed_state_index(ind, new_axes)] = tensor[ind];
+        transposed_tensor[ind] = tensor[transposed_state_index(ind, new_axes)];
```

Comment:
So the order of assignment was wrong previously?

Reply:
I remade from scratch, this part and refactor two parts of the code, this is one of them. The other one is the method `transposed_state_index`.
I didn't go too deep into why it was not working before, I have just a vague idea, but you can define it in two ways.
Like it was, and like it is now. But the indexes need to be right, and they were not always right before.